app/src/main/java/com/zulip/android/activities/ZulipActivity.java

```diff
@@ -524,6 +529,34 @@ public Cursor runQuery(CharSequence charSequence) {
                 handleSentImage(intent);
             }
         }
+
+        handleOnFragmentChange();
+        calendar = Calendar.getInstance();
+    }
+
+    private void handleOnFragmentChange() {
+        getSupportFragmentManager().addOnBackStackChangedListener(new FragmentManager.OnBackStackChangedListener() {
+            @Override
+            public void onBackStackChanged() {
+                FragmentManager fragmentManager = getSupportFragmentManager();
+                Fragment fragment = fragmentManager.findFragmentById(R.id.list_fragment_container);
+                if (fragment instanceof MessageListFragment)
```

Comment:
This `if (fragment instanceof MessageListFragment)` is a nice approach generally but in our project it is MessageListFragment only so this can be removed?